server/server.php

```diff
@@ -1,105 +1,213 @@
+                'allow_self_signed' => true,
+//                'capture_peer_cert' => false,
+            ]]);
+
+            $server
+                ->addMiddleware(new CloseHandler())
+                ->addMiddleware(new PingResponder())
+                ;
+
+            // If debug mode and logger is available
+            if (($this->options['debug'] ?? false) && class_exists('WebSocket\Test\EchoLog')) {
+                $server->setLogger(new EchoLog());
+                echo "# Using logger\n";
+            }
+            $server->setTimeout($this->options['timeout'] ?? 60);
+            echo "# Set timeout: {$this->options['timeout']}\n";
+
+            $server->setFrameSize($this->options['framesize'] ?? 4096);
+            echo "# Set frame size: {$this->options['framesize']}\n";
+
+            $server->setMaxConnections($this->options['connections'] ?? 30);
+            echo "# Set max connections: {$this->options['connections']}\n";
+
+            if ($this->options['deflate'] ?? false) {
+                $server->addMiddleware(new CompressionExtension(new DeflateCompressor()));
+                echo "# Using per-message: deflate compression\n";
+            }
+
+            $server->addMiddleware(new WebSocket\Middleware\SubprotocolNegotiation([
+                'solid-0.1'
+            ]));
+            echo "# Added subprotocol solid-0.1\n";
+
+            echo "# Listening on port {$server->getPort()}\n";
+
+            // Wiring the server handlers to our handlers;
+            $server
+                ->onHandshake(function (\WebSocket\Server $server, \WebSocket\Connection $connection, $request, $response) {
+                    return $this->onHandshake($server, $connection, $request, $response);
+                })
+                ->onDisconnect(function(\WebSocket\Server $server, \WebSocket\Connection $connection) {
+                    return $this->onDisconnect($server, $connection);
+                })
+                ->onText(function(\WebSocket\Server $server, \WebSocket\Connection $connection, \WebSocket\Message\Text $message) {
+                    return $this->onText($server, $connection, $message);
+                })
+                ->onClose(function(\WebSocket\Server $server, \WebSocket\Connection $connection, \WebSocket\Message\Close $message) {
+                    return $this->onClose($server, $connection, $message);
+                })
+                ->onError(function (\WebSocket\Server $server, \WebSocket\Connection|null $connection, ExceptionInterface $exception) {
+                    return $this->onError($server, $connection, $exception);
+                })
+            ->start();
+        } catch (\Throwable $e) {
+            echo "> ERROR: {$e->getMessage()}\n";
+        }
     }
 }
 
-$server = IoServer::factory(
-    new HttpServer(
-        new WsServer(
-            new Socket()
-        )
-    ),
-	SOCKET_PORT
-);
-
-$server->run();
+$server = new SolidPubSub($options);
```

Comment:
At some later point we might want to split the config, class, and call, but until we start adding unit-tests, I don't think it is worth the effort (yet).